File: app/src/main/java/ai/elimu/content_provider/room/entity/Content.java
package ai.elimu.content_provider.room.entity;

import androidx.annotation.NonNull;

/**
 * For documentation, see https://github.com/elimu-ai/webapp/tree/main/src/main/java/ai/elimu/model
 */
public class Content extends BaseEntity {

    @NonNull
    private Integer revisionNumber;

    private Integer usageCount;

    public Integer getRevisionNumber() {
        return revisionNumber;
    }

    public void setRevisionNumber(Integer revisionNumber) {
        this.revisionNumber = revisionNumber;
    }

    public Integer getUsageCount() {
        return usageCount;
    }

    public void setUsageCount(Integer usageCount) {
        this.usageCount = usageCount;
    }
}

File: app/src/main/java/ai/elimu/content_provider/room/entity/Content.kt
package ai.elimu.content_provider.room.entity

/**
 * For documentation, see [model](https://github.com/elimu-ai/webapp/tree/main/src/main/java/ai/elimu/model)
 */
open class Content : BaseEntity() {
    @JvmField
    var revisionNumber: Int = 0

    @JvmField
    var usageCount: Int? = null
}
